arcade/standard/bed_rush/map.xml

```diff
@@ -0,0 +1,587 @@
+        <variable var="round_state">1</variable>
+    </all>
+    <all id="countdown0-state">
+        <match-running/>
+        <variable var="round_state">0</variable>
+    </all>
+    <after id="countdown-bossbar" message="`a» `fNext round begins in {0} `a«" duration="5s">
+        <all>
+            <filter id="countdown"/>
+            <match-running/>
+        </all>
+    </after>
+    <any id="countdown">
+        <variable var="round_state">30</variable> <!-- reset2-state -->
+        <variable var="round_state">3</variable>
+        <variable var="round_state">2</variable>
+        <variable var="round_state">1</variable>
+        <variable var="round_state">0</variable>
+    </any>
+    <offset id="red-bed-broke" vector="0,10,-41">
+        <material>air</material>
+    </offset>
+    <offset id="blue-bed-broke" vector="0,10,41">
+        <material>air</material>
+    </offset>
+    <!-- Dummy offsets/actions required to make PGM track bed breaking correctly -->
+    <offset id="red-bed-broke-2" vector="0,10,-42">
+        <material>air</material>
+    </offset>
+    <offset id="blue-bed-broke-2" vector="0,10,42">
+        <material>air</material>
+    </offset>
+    <all id="should-tp">
+        <variable var="round_state">20</variable> <!-- reset1-state -->
+        <participating/>
+        <match-running/>
+        <alive/>
+    </all>
+    <deny id="original-block-protection-filter">
+        <blocks region="original-block-protection">
+            <not>
+                <any>
+                    <!-- List of blocks in the original map that players can modify -->
+                    <material>air</material>
+                    <material>bed block</material>
+                    <material>sandstone</material>
+                    <material>stained clay</material>
+                </any>
+            </not>
+        </blocks>
+    </deny>
+    <deny id="red-bed-protect-1">
+        <all>
+            <blocks region="bed-red-1">
+                <all>
+                    <material>bed block</material>
+                </all>
+            </blocks>
+            <team>team-one</team>
+        </all>
+    </deny>
+    <deny id="red-bed-protect-2">
+        <all>
+            <blocks region="bed-red-2">
+                <all>
+                    <material>bed block</material>
+                </all>
+        </blocks>
+        <team>team-one</team>
+        </all>
+    </deny>
+    <deny id="blue-bed-protect-1">
+        <all>
+            <blocks region="bed-blue-1">
+                <all>
+                    <material>bed block</material>
+                </all>
+            </blocks>
+            <team>team-two</team>
+        </all>
+    </deny>
+    <deny id="blue-bed-protect-2">
+        <all>
+            <blocks region="bed-blue-2">
+                <all>
+                    <material>bed block</material>
+                </all>
+            </blocks>
+            <team>team-two</team>
+        </all>
+    </deny>
```

Comment:
Because you know at load time those blocks will be beds, making original blocks filters makes no sense. Also, i don't really see the need for any of it. instead you can just:

1) Update the bed region definitions to be (these are defined lower below):
```xml
    <union id="bed-red">
        <block id="bed-red-1">0,10,-41</block>
        <block id="bed-red-2">0,10,-42</block>
    </union>
    <union id="bed-blue">
        <block id="bed-blue-1">0,10,41</block>
        <block id="bed-blue-2">0,10,42</block>
    </union>
```

2) Update the applied regions (defined lower below):
```xml
    <apply block-break="deny-red" region="bed-red" message="You may not break your own bed!"/>
    <apply block-break="deny-blue" region="bed-blue" message="You may not break your own bed!"/>
```

3) define some simple deny-blue and deny-red filters instead:
```xml
  <deny id="deny-red">
    <team>team-one</team>
  </deny>
  <deny id="deny-blue">
    <team>team-two</team>
  </deny>
```

Reply:
Yes, this is clearly superior solution, thank you.